omega/omega-context/src/main/java/org/apache/servicecomb/saga/omega/context/CompensationContext.java

```diff
@@ -39,14 +44,21 @@ public void addCompensationContext(Method compensationMethod, Object target) {
   public void apply(String globalTxId, String localTxId, String compensationMethod, Object... payloads) {
     CompensationContextInternal contextInternal = contexts.get(compensationMethod);
 
+    String oldGlobalTxId = omegaContext.globalTxId();
+    String oldLocalTxId= omegaContext.localTxId();
     try {
+      omegaContext.setGlobalTxId(globalTxId);
```

Comment:
Current solution is not good enough to be merged.
OmegaContext is not static,  the internal method could not use it directly.
I think we could add annotation to help the user inject the context into the method.